jwst/assign_wcs/pointing.py

```diff
@@ -223,19 +223,17 @@ def create_fitswcs(inp, input_frame=None):
         wcsinfo = wcsinfo_from_model(inp)
         wavetable = None
         spatial_axes, spectral_axes, unknown = gwutils.get_axes(wcsinfo)
-        sp_axis = spectral_axes[0]
-        if wcsinfo['CTYPE'][sp_axis] == 'WAVE-TAB':
-            wavetable = inp.wavetable
-        transform = fitswcs_transform_from_model(wcsinfo, wavetable)
+        if spectral_axes:
+            sp_axis = spectral_axes[0]
+            if wcsinfo['CTYPE'][sp_axis] == 'WAVE-TAB':
+                wavetable = inp.wavetable
+        transform = fitswcs_transform_from_model(wcsinfo, wavetable=wavetable)
         output_frame = frame_from_model(wcsinfo)
-    #elif isinstance(inp, str):
-        #transform = create_fitswcs_transform(inp)
-        #output_frame = frame_from_fits(inp)
     else:
         raise TypeError("Input is expected to be a DataModel instance or a FITS file.")
 
     if input_frame is None:
-        input_frame = "detector"
+        input_frame = cf.Frame2D(name="detector")
```

Comment:
So this is not quite correct.  As originally implemented, the string "detector" is generic and stand in for a 2D image input frame or 3D cube input frame.  If I want to actually use a `CoordinateFrame` object here instead, especially for `cube_build` which uses this function to generate the `gwcs.WCS` object for the final IFU cubes, it needs to have something like `Frame3D` coordinate frame as input, which doesn't currently exist.  Can I use `CoordinateFrame` directly?  @nden

Reply:
Or should I just switch it back to the string input frame, as it was?

Reply:
I implemented a solution that checks WCSAXES keyword and uses `Frame2D` or `CoordinateFrame(naxes=3)` appropriately.  And fixed up the tests.